int16 theta
int16 r
int16 z
